include/seqan3/alignment/data_structure/aligned_sequence_concept.hpp

```diff
@@ -0,0 +1,258 @@
+ * \extends   seqan3::forward_range_concept
+ * \brief     The generic concept for an aligned sequence.
+ * \ingroup   data_structure
+ *
+ * \details
+ *
+ * This concept describes the requirements a sequence must fulfill
+ * in order to be part of the seqan3::alignment object.
+ */
+/*!\name Requirements for seqan3::aligned_sequence_concept
+ * \brief You can expect these functions on all types that model the seqan3::aligned_sequence_concept.
+ *
+ * \details
+ *
+ * Type Requirements:
+ *   1. The type must module the seqan3::forward_range_concept.
+ *   2. The `value_type` must module the alphabet concept.
+ *   3. The `value_type` must be assignable from seqan3::gap::GAP.
+ *
+ * Furthermore, the free functions seqan3::aligned_sequence_concept::insert_gap
+ * and seqan3::aligned_sequence_concept::erase_gap exist for the type.
+ * \{
+ */
+/*!\fn      inline void insert_gap(seq_type & seq, typename seq_type::const_iterator pos_it)
+ * \brief   Insert a seqan3::gap::GAP into a sequence container.
+ * \relates seqan3::aligned_sequence_concept
+ * \tparam        seq_type The sequence container type. Must model the
+ *                         seqan3::sequence_container_concept. It's
+ *                         `value_type` must be assignable from seqan3::gap::GAP.
+ * \param[in,out] seq      The aligned sequence container to modify.
+ * \param[in]     pos_it   The iterator pointing to the position of where to insert a gap.
+ *
+ * \details
+ * \attention This is a concept requirement, not an actual function (however types
+ *            modeling this concept will provide an implementation).
+ */
+/*!\fn      inline void insert_gap(seq_type & seq, typename seq_type::const_iterator pos_it, typename seq_type::size_type count)
+ * \brief   Insert seqan3::gap::GAP 's into a sequence container.
+ * \relates seqan3::aligned_sequence_concept
+ * \tparam        seq_type The sequence container type. Must model the
+ *                         seqan3::sequence_container_concept. It's
+ *                         `value_type` must be assignable from seqan3::gap::GAP.
+ * \param[in,out] seq      The aligned sequence container to modify.
+ * \param[in]     pos_it   The iterator pointing to the position of where to insert a gap.
+ * \param[in]     count    The number of gaps to insert.
+ *
+ * \details
+ * \attention This is a concept requirement, not an actual function (however types
+ *            modeling this concept will provide an implementation).
+ */
+/*!\fn      inline void erase_gap(seq_type & seq, typename seq_type::const_iterator pos_it)
+ * \brief   Erase a gap from a sequence container.
+ * \relates seqan3::aligned_sequence_concept
+ *
+ * \tparam        seq_type The sequence container type. Must model the
+ *                         seqan3::sequence_container_concept. It's `value_type`
+ *                         must be assignable from seqan3::gap::GAP.
+ * \param[in,out] seq      The aligned sequence container to modify.
```

Comment:
Why did my comments here dissappear? In any case, this should just be forward_range...